### PR TITLE
Use red or green background for whitespace-only diffs

### LINKED ISSUE #5191: Whitespace differences are not highlighted by ExUnit
### Environment
- Elixir version (elixir -v):

```
Erlang/OTP 18 [erts-7.3] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.3.2
```
- Operating system:

OS X El Capitan
### Current behavior

Given an assertion like:

``` elixir
assert [%{a: "abc "}, %{a: "def"}, %{c: "gh"}] == [%{a: "abc"}, %{a: " def"}, %{c: "ghi"}]
```

ExUnit currently produces a diff like:

screen shot 2016-09-07 at 2 17 38 pm
### Expected behavior

This diff is confusing.  It looks like the only difference is the addition of the letter `i` but there are actually some whitespace differences, too. Since those are not highlighted, my brain is drawn to the highlighted `i` and I initially wrongly interpreted it as meaning that the `i` was the only difference.  The diff would be less-confusing and more useful if whitespace differences were highlighted, too.  As an example, here's how `git diff` shows whitespace differences:

screen shot 2016-09-07 at 1 49 12 pm

(See also #4689 for some prior art that was later reverted.)

### PR DESCRIPTION
**Pull request by josevalim**
Closes #5191.

### CODE REVIEW

**Comment by josevalim** [comment]
screen shot 2016-11-21 at 19 07 51

File: lib/ex_unit/examples/difference.exs [hunk 1]
```diff
@@ -25,6 +25,12 @@ defmodule Difference do
     assert string1 == string2
   end
 
+  test "whitespace" do
+    list1 = [%{a: "abc "}, %{a: "def"}, %{c: "gh"}]
+    list2 = [%{a: "abc"}, %{a: " def"}, %{c: "ghi"}]
```
Comment (lexmag) [inline lib/ex_unit/examples/difference.exs]: We can use `"hi"` for last string here, so we get `:diff_delete` color for previous line.

**Comment by josevalim** [comment]
Now with slightly brighter colors:

screen shot 2016-11-21 at 19 28 36

I believe I prefer the original one because this one looks too close to the regular diff and it does not send a message it is a whitespace diff.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**